### PR TITLE
fix(Tabs): prevent update selecton upon arrow keys

### LINKED ISSUE #4870: [Tabs, Content switcher] Manual activation for accessible users
Manual activation missing on Tab and Content Switcher

## What package(s) are you using? 
Current version of React

## Detailed description

It seems as though Tabs and Content Switcher do not provide a way to support Manual Activation. This causes UX issues for accessible users. The system loads the contents of each section as the user navigates between them, even if they are not interested in that content.

> Is this issue related to a specific component?
Tabs, Content Switcher

> What did you expect to happen? What happened instead? What would you like to
> see changed?
Implement Manual Activation

> What browser are you working in?
Chrome, Firefox, Safari

> What version of the Carbon Design System are you using?
Current

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?
IoT Maximo Suite, release Q2 2020

### LINKED ISSUE #5657: Content switcher and tab selection should be based on explicit selection instead of on navigation
From @aagonzales at https://github.com/carbon-design-system/carbon/issues/5580#issuecomment-598309198:

> We'd want tabpanel content to be shown when the button is "pressed" so that the user can easily cycle through the tab options before reading/navigation the tabpanel content.

From @carmacleod in the same issue:

> But there's a bit more to it than that...  ;)
Here's the [tabpanel pattern in the APG](https://w3c.github.io/aria-practices/#tabpanel) (also called _Tabs_).
> Please read through all of section _3.23 Tabs_ (including the _Examples_, _Keyboard Interaction_, and _ARIA Roles, States and Properties_ subsections) and decide if it's what you are creating.
> If you go with this pattern, please carefully re-read that section about WAI-ARIA Roles, States, and Properties to make sure you get the aria markup just right, because screen readers are picky. ;)

> The 2 example pages show slightly different keyboard handling, depending on whether or not the tabpanel content is shown when the button is "selected (focused)" or when it is "pressed" - you decide which one to use after reading the pattern information (fyi, your codepen example works like the "Manual Activation" one, which may or may not be what you want for your production content switcher). Remember to add arrow key handling  for the buttons if you are going with this pattern (i.e. _instead of_ using tab key; i.e. the buttons need `tabindex="-1"` when they're not selected). Screen reader users will expect arrows to work as soon as they hear they are in a tablist or tabpanel, and keyboard users will know to use arrow keys to change the content, and they will expect that using the tab key navigates to the content instead of tabbing through all of the tabs in the tablist before getting to the content.

> After trying out the examples, please scroll further down the example page to see the actual Keyboard Support for each example, followed by the Role, Property, State, and Tabindex Attributes used in the example, followed by the Source Code.

### PR DESCRIPTION
**Pull request by asudoh**
This change makes left/right arrow keys in `<ContentSwitcher>` and `<Tabs>` work for change only the focused item, not the selection. User can use space/enter keys to update the selection.

Fixes #5657.
Fixes #4870.

#### Changelog

**Removed**

- Code that selects `<Switch>`/`<Tab>` upon arrow key navigation.

#### Testing / Reviewing

Testing should make sure `<ContentSwitcher>`/`<Tabs>` are not broken.

### CODE REVIEW

**Review by emyarod — APPROVED** [review]
content switcher and tabs now have the same keyboard nav behavior, looks good to me

**Comment by aagonzales** [comment]
Will a separate PR take into account the other behavior variation as mentioned in this comment?
https://github.com/carbon-design-system/carbon/issues/5657#issuecomment-601282499

**Comment by asudoh** [comment]
> Will a separate PR take into account the other behavior variation as mentioned in this comment?
#5657 (comment)

@aagonzales Thank you for jumping in and commenting. Wanted to double-check on this... It means that you still want a current behavior (switch-tab-on-focus). Is it correct? If so, may we assume that it's application-level choice (i.e. `<Tabs switchOnFocus={true}>`), instead of end-user-level choice (i.e. some user gesture switches one mode vs. another)?

**Comment by aagonzales** [comment]
Yes, we want to allow product teams to chose between the two behaviors. Both behaviors have positive and negatives and the scenario and content will dictate which is best to use. So we want to keep the current behavior but also offer the other.

**Comment by asudoh** [comment]
Thank you for clarifying @aagonzales! Brought the old behavior back and implemented the new behavior as an option, that is enabled by `<Tabs requiresExplicitSelection={true}>`.

**Review by aagonzales — COMMENTED** [review]
These seems to be a little lag with the focus state of the selected tab. The bar disappears with a visible lag before the focus appears.

Mar-24-2020 09-27-39

**Comment by emyarod** [comment]
IIRC that's a storybook UI related issue since it doesn't occur in the iframe https://deploy-preview-5659--carbon-components-react.netlify.com/iframe.html?id=tabs--container

**Review by aagonzales — APPROVED** [review]
Ok looks good then.

File: packages/react/__tests__/__snapshots__/PublicAPI-test.js.snap [hunk 1]
```diff
@@ -660,6 +660,9 @@ Map {
         "isRequired": true,
         "type": "func",
       },
+      "requiresExplicitSelection": Object {
```
Comment (joshblack) [inline packages/react/__tests__/__snapshots__/PublicAPI-test.js.snap]: Would it help if we named this something like `focusMode`? Could be `PropTypes.oneOf(['automatic', 'manual'])`

Comment (aagonzales) [reply packages/react/__tests__/__snapshots__/PublicAPI-test.js.snap]: oh yeah I like that!

Comment (asudoh) [reply packages/react/__tests__/__snapshots__/PublicAPI-test.js.snap]: Good suggestion - Used `selectionMode` to better reflect what the new user interaction pattern does.

**Comment by abbeyhrt** [comment]
Would this close #4870 as well?

**Review by abbeyhrt — APPROVED** [review]
Looks good to me! 🎉

**Comment by asudoh** [comment]
Great point @abbeyhrt! Added #4870 as a to-be-fixed issue.

**Pull request merged**